src/util.ts

```diff
@@ -89,19 +89,59 @@ export async function signAddress(string: string, privateKey: string) {
 
 /**
  * Generates a random string of the specified length
+ *  need to avoid bias: https://gist.github.com/joepie91/7105003c3b26e65efcea63f3db82dfba#so-how-do-i-obtain-random-values-securely
+ * h/t to Nanak Nihal from Holonym
+ * browser: tries to use secure generateKey if available, otherwise falls back to getRandomValues
+ * node: uses secure crypto.randomBytes
  */
-export function getRandomString(n: number = 16): string {
+export async function getRandomString(n: number = 16): Promise<string> {
 	const charset = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
 	const charsetLength = charset.length
 	const maxByteValue = 256 // Each byte has 256 possible values
 	const maxUnbiasedByte = maxByteValue - (maxByteValue % charsetLength)
 
-	// need to avoid bias: https://gist.github.com/joepie91/7105003c3b26e65efcea63f3db82dfba#so-how-do-i-obtain-random-values-securely
-	// h/t to Nanak Nihal from Holonym
-
 	let randomString = ''
+
+	const generateKeyRandomBytes = async (length: number): Promise<Uint8Array> => {
+		const cryptoSubtle =
+			typeof window !== 'undefined' && window.crypto && window.crypto.subtle ? window.crypto.subtle : undefined
+		if (cryptoSubtle) {
+			try {
+				// Use generateKey to generate a symmetric key of sufficient length
+				const key = await cryptoSubtle.generateKey(
+					{
+						name: 'AES-GCM',
+						length: length * 8, // Convert byte length to bit length
+					},
+					true,
+					['encrypt', 'decrypt']
+				)
+				// Export the key to raw bytes
+				const keyBuffer = await cryptoSubtle.exportKey('raw', key)
+				return new Uint8Array(keyBuffer)
+			} catch (error) {
+				console.warn('Failed to use generateKey. Falling back to getRandomValues.', error)
+			}
+		}
+		return getRandomValuesRandomBytes(length)
+	}
+
+	const getRandomValuesRandomBytes = (length: number): Uint8Array => {
+		if (typeof window !== 'undefined' && window.crypto && window.crypto.getRandomValues) {
+			// Browser
+			const array = new Uint8Array(length)
+			window.crypto.getRandomValues(array)
+			return array
+		} else {
+			// Node
```

Comment:
What will happen if the user is on mobile? not every mobile browser has window.crypto defined, so will the function land here? Will the 'require' not fail on mobile?